src/librustc/dep_graph/graph.rs

```diff
@@ -18,6 +18,7 @@ use std::env;
 use std::hash::Hash;
 use ty::{self, TyCtxt};
 use util::common::{ProfileQueriesMsg, profq_msg};
+use serialize::{Decodable, Decoder};
```

Comment:
Why are new imports needed here now? This looks like an odd change.

Reply:
This is due to the `newtype_index` changes

Reply:
Can't `newtype_index` be written in a such a way that this isn't needed?

Reply:
Some crates import `rustc_serialize` as `rustc_serialize` and some import it as `serialize`. So I can't just use global paths.

I also can't add my own `extern crate rustc_serialize;`... unless I put the trait impls into e.g. the `new` method where it won't conflict with other things in scope

Reply:
done, that worked